nexus/db-model/src/multicast_group.rs

```diff
@@ -0,0 +1,483 @@
+/// following the pattern used throughout Omicron where the user-facing resource
+/// uses the simpler name. External multicast groups are the primary user-facing
+/// multicast resources, while underlay groups are internal infrastructure.
+pub type MulticastGroup = ExternalMulticastGroup;
+
+/// An external multicast group for delivering packets to multiple recipients.
+///
+/// External groups are multicast groups allocated from IP pools. These are
+/// distinct from [UnderlayMulticastGroup] which are system-generated IPv6 addresses for
+/// NAT mapping.
+#[derive(
+    Queryable,
+    Selectable,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Resource,
+    Serialize,
+    Deserialize,
+)]
+#[diesel(table_name = multicast_group)]
+pub struct ExternalMulticastGroup {
+    #[diesel(embed)]
+    pub identity: ExternalMulticastGroupIdentity,
+    /// IP pool this address was allocated from.
+    pub ip_pool_id: Uuid,
+    /// IP pool range this address was allocated from.
+    pub ip_pool_range_id: Uuid,
+    /// VNI for multicast group (derived or random).
+    pub vni: Vni,
+    /// Primary multicast IP address (overlay/external).
+    pub multicast_ip: IpNetwork,
+    /// Source IP addresses for Source-Specific Multicast (SSM).
+    /// Empty array means any source is allowed.
+    pub source_ips: Vec<IpNetwork>,
+    /// Multicast VLAN (MVLAN) for egress multicast traffic to upstream networks.
+    ///
+    /// When specified, this VLAN ID is passed to switches (via DPD) as part of
+    /// the `ExternalForwarding` configuration to tag multicast packets leaving
+    /// the rack. This enables multicast traffic to traverse VLAN-segmented
+    /// upstream networks (e.g., peering with external multicast sources/receivers
+    /// on specific VLANs).
+    ///
+    /// The MVLAN value is sent to switches during group creation/updates and
+    /// controls VLAN tagging for egress traffic only; it does not affect ingress
+    /// multicast traffic received by the rack. Switch port selection for egress
+    /// traffic remains pending (see TODO at `nexus/src/app/multicast/dataplane.rs:113-115`).
+    ///
+    /// Valid range when specified: 2-4094 (IEEE 802.1Q; Dendrite requires >= 2).
+    ///
+    /// Database Type: i16 (INT2) - this field uses `i16` (INT2) for storage
+    /// efficiency, unlike other VLAN columns in the schema which use `SqlU16`
+    /// (forcing INT4). Direct `i16` is appropriate here since VLANs fits in
+    /// INT2's range.
+    pub mvlan: Option<i16>,
+    /// Associated underlay group for NAT.
+    /// Initially None in ["Creating"](MulticastGroupState::Creating) state,
+    /// populated by reconciler when group becomes ["Active"](MulticastGroupState::Active).
+    pub underlay_group_id: Option<Uuid>,
+    /// Rack ID multicast group was created on.
+    pub rack_id: Uuid,
+    /// Group tag for lifecycle management.
+    pub tag: Option<String>,
```

Comment:
Looks like this is for managing tagged dendrite elements. I'd update the comment to add that as I was not sure what lifecycle management meant without looking for how this field is used.

Dendrite tags are also being used a bit differently for multicast. Existing tag usage has been to indicate what entity created a bit of state in dpd. That could be something like "omicron", "ddm", or "mgd", Here it looks like a tag is used for grouping coupled underlay/overlay multicast state in dendrite on a per group basis. That makes sense, but I think it's spelling out in a comment somewhere.